### PR TITLE
[v12] Change error response formatting for "/version" endpoint

### PR DESCRIPTION
**Pull request by tigrato**
Backport #21289 to branch/v12

### CODE REVIEW

File: lib/client/conntest/kube.go [hunk 1]
```diff
@@ -257,8 +257,10 @@ func (s KubeConnectionTester) handleErrFromKube(ctx context.Context, clusterName
 		// agents is still running an older version.
 		// For this reason, messages are not shared between this connection test and
 		// `kubernetes_service` to force detection of incompatible messages.
+		// TODO(tigrato): Remove this check once we no longer support Teleport versions bellow 12.
```
Comment (jakule) [inline lib/client/conntest/kube.go:260]: This should be marked as `DELETE IN 14.0` if I understand this correctly.

File: lib/kube/proxy/forwarder.go [hunk 2]
```diff
@@ -255,6 +255,14 @@ func NewForwarder(cfg ForwarderConfig) (*Forwarder, error) {
 
 	fwd.router.UseRawPath = true
 
+	fwd.router.GET("/version", fwd.withAuth(
+		func(ctx *authContext, w http.ResponseWriter, r *http.Request, _ httprouter.Params) (interface{}, error) {
```
Comment (jakule) [inline lib/kube/proxy/forwarder.go:259]: nit: (We're using Go 1.20 on all production branches thanks to @codingllama)
```suggestion
		func(ctx *authContext, w http.ResponseWriter, r *http.Request, _ httprouter.Params) (any, error) {
```

File: lib/kube/proxy/forwarder.go [hunk 3]
```diff
@@ -635,9 +673,16 @@ func (f *Forwarder) setupContext(authCtx auth.Context, req *http.Request, isRemo
 	if !isRemoteCluster {
 		// check signing TTL and return a list of allowed logins for local cluster based on Kubernetes service labels.
 		kubeAccessDetails, err := f.getKubeAccessDetails(roles, kubeCluster, sessionTTL, kubeResource)
-		if err != nil {
+		if err != nil && !trace.IsNotFound(err) {
 			return nil, trace.Wrap(err)
+			// roles.CheckKubeGroupsAndUsers returns trace.NotFound if the user does
+			// does not have at least one configured kubernetes_users or kubernetes_groups.
+		} else if trace.IsNotFound(err) {
+			errMsg := "Your user's Teleport role does not allow Kubernetes access." +
```
Comment (jakule) [inline lib/kube/proxy/forwarder.go:681]: nit:
```suggestion
			const errMsg = "Your user's Teleport role does not allow Kubernetes access." +
```

**Review by jakule — APPROVED** [review]
I wasn't part of the original review, hence a few nit comments. I do not mind if you ignore them as this is backport.

File: lib/kube/proxy/forwarder.go (hunk 2 above)
Comment (codingllama) [reply lib/kube/proxy/forwarder.go:259]: Not on v10 yet, I keep having issues with golangci locally. :/

**Comment by codingllama** [comment]
My 2c regarding nits on backports: if you do take those, take them in a PR against master and then backport the nits too.

**Review by capnspacehook — APPROVED** [review]

**Pull request merged**